src/common/utils/utils.go

```diff
@@ -181,3 +183,15 @@ func StructContainsProperty(s interface{}, property string) (bool, reflect.Value
 
 	return true, field
 }
+
+func CreateTagsForBlock(tagGroup tagging.ITagGroup, block structure.IBlock) {
```

Comment:
@tronxd This needs to be tag group sepcific, as the value passed to `tag.CalculateValue()` is different between tag groups. We can make it more generic, by putting most of the logic in the `tagging.TagGroup` struct and overriding only the value sent to the tag. Which code is it saving us?